test/extensions/filters/http/lua/lua_integration_test.cc

```diff
@@ -974,6 +995,25 @@ name: lua
   testRewriteResponse(FILTER_AND_CODE);
 }
 
+// Rewrite response buffer, without original upstream response body.
+TEST_P(LuaIntegrationTest, RewriteResponseBufferWithoutUpstreamBody) {
+  const std::string FILTER_AND_CODE =
+      R"EOF(
+name: lua
+typed_config:
+  "@type": type.googleapis.com/envoy.extensions.filters.http.lua.v3.Lua
+  inline_code: |
+    function envoy_on_response(response_handle)
+      local content_length = response_handle:body():setBytes("ok")
+      response_handle:logTrace(content_length)
+
+      response_handle:headers():replace("content-length", content_length)
+    end
+)EOF";
+
+  testRewriteResponseWithoutUpstreamBody(FILTER_AND_CODE);
```

Comment:
While you're here, can we refactor this a bit to have e.g. `expectResponseBodyRewrite()`, parameterized with the body from upstream (this can be empty or filled (buffered)) and the expected rewritten body? 

So we can reuse this to test both cases. 

Thank you!